FirebaseRemoteConfig/Tests/SwiftUnit/ConfigValueOrigTests.swift

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@testable import FirebaseRemoteConfig
+import XCTest
+
+class RemoteConfigValueOrigTests: XCTestCase {
+  func testConfigValueWithDifferentValueTypes() throws {
+    let valueA = "0.33333"
+    let dataA = try XCTUnwrap(valueA.data(using: .utf8))
+
+    let configValueA = RemoteConfigValue(data: dataA, source: .remote)
+    XCTAssertEqual(configValueA.stringValue, valueA)
+    XCTAssertEqual(configValueA.dataValue, dataA)
+    XCTAssertEqual(
+      configValueA.numberValue,
+      NSNumber(floatLiteral: 0.33333)
+    )
+    XCTAssertEqual(configValueA.boolValue, (valueA as NSString).boolValue)
```

Comment:
Why is `as NSString` needed, when it wasn't in Objective-C?  If it is, it seems like an API implementation issue.

Reply:
The test string was defined as `NSString` in ObjC which has a [`boolValue` API](https://developer.apple.com/documentation/foundation/nsstring/1409420-boolvalue). Swift's `String` doesn't have the boolValue API so I did the cast.

https://github.com/firebase/firebase-ios-sdk/blob/b79b6d8573d2201bea02909b32413bdf3f9571c6/FirebaseRemoteConfig/Tests/Unit/RCNConfigValueTest.m#L26